cmd/skaffold/app/cmd/dev.go

```diff
@@ -38,7 +38,7 @@ func NewCmdDev(out io.Writer) *cobra.Command {
 	}
 	AddRunDevFlags(cmd)
 	cmd.Flags().BoolVar(&opts.TailDev, "tail", true, "Stream logs from deployed objects")
-	cmd.Flags().StringVar(&opts.Trigger, "trigger", "polling", "How are changes detected? (polling or manual)")
+	cmd.Flags().StringVar(&opts.Trigger, "trigger", "polling", "How are changes detected? (polling, manual or notify)")
```

Comment:
"notify" is rather specific to Go and Linux.  I can't think of a better term, but perhaps we can add some more text to describe it?  It's also odd that we use "trigger" here but the related options are "watch".

> How are changes detected? (polling, manual or notify)  _polling_ periodically looks for new files (see `--watch-poll-interval`), whereas _notify_ leverages OS filesystem notifications (more efficient).  _manual_ waits for keyboard input to trigger a new build.

Reply:
Some documentation would be good, yes.

The idea is that Skaffold knows all the dependencies for an artifact and can compute a "snapshot" of those dependencies (mostly last mod time). Now, to actually recompute a snapshot, do a diff and maybe trigger a rebuild/redeploy, it needs a trigger.
That trigger can be either:
 + a timer, that runs by default every 500ms. but then Skaffold has to compute the list of deps and the mod times every 500ms
 + a manual trigger. The user presses a key when they want Skaffold to compute the diff
 + a inotify based file watcher that watches the workspace as a whole and tells Skaffold.

iNotify based watchers are not good at watching folders and files that might be disapear. but it's quite good at recursively watching a top level folder that's always there.

By combining the snapshot mechanism and a inotify trigger, we get the best of both world.

IDE integration would work the same: The IDE would detect changes, notify Skaffold that "something" changed, no details. Then Skaffold would run a diff

I all cases, Skaffold is capable of debouncing rapid streams of triggers so that it behaves well when a lot of files are changed in a short period of time.

Reply:
I'd like to see this documented so users know what the expected behavior is for each option